pkg/kclient/kclient.go

```diff
@@ -63,3 +69,154 @@ func CreateObjectMeta(name, namespace string, labels, annotations map[string]str
 
 	return objectMeta
 }
+
+func (c *Client) CreateTLSSecret(tlsCertificate []byte, tlsPrivKey []byte, componentName string, applicationName string, portNumber int) (*corev1.Secret, error) {
```

Comment:
please move these functions to a new file like ingress.go

Reply:
updated